### PR TITLE
Server Comment Functionality

### PR DESCRIPTION
**Pull request by dterr**
Backlog of commit. This shows the MVP for comment functionality. Future commits will be spaced out and single-scoped.

### CODE REVIEW

**Comment by dterr** [comment]
View the live website here:
https://dterr-step-2020.uc.r.appspot.com

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -14,19 +14,74 @@
...
+      datastore.put(commentEntity);
+    }
+    response.sendRedirect("/contact.html");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (mwo-2020) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Is this function used?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -13,16 +13,23 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Fetches the current state of the game and builds the UI.
```
Comment (mwo-2020) [inline portfolio/src/main/webapp/script.js]: Update the comment

File: portfolio/src/main/webapp/contact.html [hunk 3]
```diff
@@ -14,13 +14,22 @@
...
         <ul>
           <li><a href="https://www.linkedin.com/in/dominicterrones/">My LinkedIn</a></li>
           <li><a href="https://github.com/dterr">Dominic Terrones on GitHub</a></li>
           <li><a href="mailto:dterrones31@gmail.com"> Email me! </a></li>
         </ul>
+      <h2> Leave a Comment! </h2>
+      <form action="/data" method="POST">
+        <input type="text" id="author" name="author" placeholder="Your Name" required>
+        <br>
+        <textarea rows="5" cols="60" id="comment-message" name="comment-message" placeholder="Your Message"></textarea>
```
Comment (mwo-2020) [inline portfolio/src/main/webapp/contact.html]: Is it possible to make the comment also required?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (dterr) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Just tested, apparently not.

**Comment by dterr** [comment]
Redeployed server.

**Review by mwo-2020 — APPROVED** [review]

**Pull request merged**